include/RED4ext/Handle.hpp

```diff
@@ -125,25 +30,28 @@ class Handle : public HandleBase
 
     Handle(const Handle& aOther) noexcept
     {
-        aOther.IncRef();
-        instance = aOther.instance;
-        refCount = aOther.refCount;
+        this->CopyConstructFrom(aOther);
     }
 
     Handle(Handle&& aRhs) noexcept
     {
-        MoveConstructFrom(std::move(aRhs));
+        this->MoveConstructFrom(std::move(aRhs));
     }
 
     Handle(const WeakHandle<T>& aOther)
     {
-        if (!TryConstructFromWeak(aOther))
-            throw std::bad_weak_ptr();
+        this->TryCopyConstructFromWeak(aOther);
     }
 
     ~Handle()
     {
-        DecRef();
+        static_assert(std::is_base_of_v<ISerializable, T>,
```

Comment:
I placed it in dtor to allow handles with incomplete types. 
```c++
struct IRenderResourceBlob;
struct CMesh : res::StreamedResource
{
    Handle<IRenderResourceBlob> renderResourceBlob;
```
There's a lot of them in generated.
But assert in dtor will fail only when you try to actually use something.

Reply:
Are you sure about this? I remember that `static_asserts` at compile time. If that's a trick to get this done I'm up for it, but I want to understand it a bit.

Reply:
The main problem is that trait `std::is_base_of_v` always fails for incomplete types. It makes sense for checking inheritance/compatibility, you need the full type.
When the type is used as a property or as a parameter in function declaration, only asserts placed right in the body of the type declaration will be executed (also new `requires` clause). But asserts placed in methods are performed only when method is used. I choose the dtor because it's always needed for our own code. It's not perfect, but covers the most important case: when we control the lifetime.
To enforce the full check we have to replace forward declarations with includes.

Reply:
Thanks for the explanation! Today I learned something new 😃